src/core/js/adapt.js

```diff
@@ -74,19 +74,21 @@ class AdaptSingleton extends LockingModel {
 
   /**
    * wait until there are no outstanding completion checks
-   * @param {Function} callback Function to be called after all completion checks have been completed
+   * @param {Function} [callback] Function to be called after all completion checks have been completed
    */
-  deferUntilCompletionChecked(callback) {
-    if (this.get('_outstandingCompletionChecks') === 0) return callback();
-
-    const checkIfAnyChecksOutstanding = (model, outstandingChecks) => {
-      if (outstandingChecks !== 0) return;
-      this.off('change:_outstandingCompletionChecks', checkIfAnyChecksOutstanding);
-      callback();
-    };
-
-    this.on('change:_outstandingCompletionChecks', checkIfAnyChecksOutstanding);
-
+  async deferUntilCompletionChecked(callback) {
```

Comment:
Could set default parameter to remove the need for existence checks:
```js
callback = () => {}
```

Reply:
I did do that originally... But I'm not sure, feels a bit weird. Are you insistent or ambivalent?

Reply:
> Are you insistent or ambivalent?

My already-approved review would imply simple ambivalence.

Reply:
Duly noted.